android/app/src/googleplay/java/org/fossasia/openevent/utils/map/ClusterRenderer.java

```diff
@@ -0,0 +1,95 @@
+    private GoogleMap map;
+    private MicrolocationClusterWrapper microlocationClusterWrapper;
+
+    public ClusterRenderer(Context context, GoogleMap map,
+                             ClusterManager<MicrolocationClusterWrapper> clusterManager) {
+        super(context, map, clusterManager);
+        this.context = context;
+        this.map = map;
+        setOnClusterItemClickListener(this);
+    }
+
+    @Override
+    protected void onBeforeClusterItemRendered(MicrolocationClusterWrapper item, MarkerOptions markerOptions) {
+        super.onBeforeClusterItemRendered(item, markerOptions);
+
+        markerOptions.title(item.getMicrolocation().getName());
+        if (microlocationClusterWrapper != null && item.equals(microlocationClusterWrapper)) {
+            markerOptions.icon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.color_primary));
+        } else {
+            markerOptions.icon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.dark_grey));
+        }
+    }
+
+    @Override
+    protected void onClusterItemRendered(final MicrolocationClusterWrapper clusterItem, Marker marker) {
+        super.onClusterItemRendered(clusterItem, marker);
+        clusterItem.setMarker(marker);
+   }
+
+    public void destroyContext() {
+       context = null;
+   }
+
+    @Override
+    public boolean onClusterItemClick(MicrolocationClusterWrapper item) {
+        if (microlocationClusterWrapper != null) {
+            getMarker(microlocationClusterWrapper).setIcon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.dark_grey));
+        }
+        microlocationClusterWrapper = item;
+        getMarker(item).setIcon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.color_primary));
+        return false;
+    }
+
+    public void focusOnMarkers(Map<String, MicrolocationClusterWrapper> stringMicrolocationClusterWrapperMap, String locationName) {
+        map.animateCamera(CameraUpdateFactory.newLatLngZoom(stringMicrolocationClusterWrapperMap.get(locationName).getPosition(), map.getMaxZoomLevel()), new GoogleMap.CancelableCallback() {
+            @Override
+            public void onFinish() {
+                Handler handler = new Handler();
+                handler.postDelayed(() -> {
+                    MicrolocationClusterWrapper item = stringMicrolocationClusterWrapperMap.get(locationName);
+                    Marker marker = item.getMarker();
+                    if (microlocationClusterWrapper != null)
+                        getMarker(microlocationClusterWrapper).setIcon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.dark_grey));
+                    microlocationClusterWrapper = item;
+                    marker.showInfoWindow();
+                    marker.setIcon(ImageUtils.vectorToBitmap(context, R.drawable.map_marker, R.color.color_primary));
+                }, 100);
+            }
+
+            @Override
+            public void onCancel() {
```

Comment:
Add comment here like "// do nothing" or something appropriate. 
As a best practice we always try to reduce codacy issue. Every time if there is codacy issue than try to resolve it. Thanks :smile: